src/modules/logcollector/src/logcollector.cpp

```diff
@@ -101,6 +118,40 @@ void Logcollector::AddReader(std::shared_ptr<IReader> reader) {
     EnqueueTask(reader->Run());
 }
 
+void Logcollector::CleanAllReaders() {
+    for (const auto &reader : m_readers) {
+        reader->Stop();
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(m_timersMutex);
+        for (const auto &timer : m_timers) {
+            timer->cancel();
+        }
+    }
+
+    while (m_activeReaders) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(10)); // NOLINT
+    }
+    m_readers.clear();
+}
+
 Awaitable Logcollector::Wait(std::chrono::milliseconds ms) {
-    co_await boost::asio::steady_timer(m_ioContext, ms).async_wait(boost::asio::use_awaitable);
+    if (!m_ioContext.stopped()) {
+        auto timer = boost::asio::steady_timer(m_ioContext, ms);
+        {
+            std::lock_guard<std::mutex> lock(m_timersMutex);
+            m_timers.push_back(&timer);
+        }
+        try {
+            co_await timer.async_wait(boost::asio::use_awaitable);
```

Comment:
If there are other reasons why this throws I'd change the message at least slightly so it is not misleading...